### DISCUSSION TITLE
How to maintain css transitions in custom cell whenever data changes

### DISCUSSION THREAD

**Discussion by Newbie012** [TanStack/table#3085] [Q&A]
Basically, I'm trying to think of a way to maintain the CSS transitions in custom cells when the data of the table is changed. I'll give you an example:

I have a table of posts. Each post has an "active" column. This column is a UI switch. When the value of the cell (boolean) is true, the switch is turned on, and when it's off, then it toggles off. Whenever I toggle on or off the switch, there's a small animation that occurs, but not in react-table and I think I know why.

I assume, that whenever the data of the table is changed, then each `Cell` callback is being invoked and it returns a whole new component each time. If I'm wrong (hopefully), what would be the best approach here?

## TL;DR - "Why it's not working"

I have created a demo here - https://codesandbox.io/s/brave-mendeleev-lkf5d?file=/src/index.css

### This is the expected outcome:

[![A box with a background transition][2]][2]

### This is the actual outcome:

[![a table with boxes that won't transition][3]][3]

  [1]: https://www.npmjs.com/package/react-table
  [2]: https://i.stack.imgur.com/Rk4Vd.gif
  [3]: https://i.stack.imgur.com/Jg9UN.gif

**Comment by chukwudobemicah** [comment]
I also have this issue, whats the fix here?

**Reply by Newbie012** [reply]
I gave up

**Comment by lerman01** [comment]
I've encountered the same issue as well. It's been 4 years and still no answer, so I guess this is a framework limitation?
